### PR TITLE
Accept header negotiation

### PR DESCRIPTION
**Pull request by philcleveland**
Adds initial implementation of Accept header parsing and prioritization. Based on http://www.w3.org/Protocols/rfc2616/rfc2616-sec14.html

### CODE REVIEW

File: negotiate.go [hunk 1]
```diff
@@ -22,11 +22,24 @@ func New(responseProcessors ...*ResponseProcessor) {
...
 //Negotiate your model based on HTTP Accept header
 func Negotiate(w http.ResponseWriter, req *http.Request, model interface{}) {
-	for _, processor := range processors {
-		acceptHeader := req.Header.Get("Accept")
-		if processor.CanProcess(acceptHeader) {
-			processor.Process(w, model)
-			return
+
+	accept := new(Accept)
+	//TODO:test should not be case sensitive
```
Comment (jchannon) [inline negotiate.go:27]: What's this for? Headers are case-sensitive

**Comment by philcleveland** [comment]
The actual word "Accept".  I am not seeing anything in rfc2616 that indicates that "accept" should not work.  Did I just miss it?  NancyFX also has a test "Should_ignore_case_of_accept_header_name_when_retrieving_values" which made me think it should not be case sensitive.

**Comment by jchannon** [comment]
Hmm for some reason I thought it was. If in doubt ask Glenn Miller in these
situations :)

**Comment by philcleveland** [comment]
Who is Glenn Miller?  Is that a mash up of Glenn Block and Darrel Miller? LOL

**Comment by jchannon** [comment]
Yup :) watch their first video podcast of http

**Pull request merged**